Mathlib/Algebra/Star/Free.lean

```diff
@@ -77,10 +75,12 @@ theorem star_algebraMap (r : R) : star (algebraMap R (FreeAlgebra R X) r) = alge
   simp [star, Star.star]
 #align free_algebra.star_algebra_map FreeAlgebra.star_algebraMap
 
-/-- `star` as an `alg_equiv` -/
-def starHom : FreeAlgebra R X ≃ₐ[R] (FreeAlgebra R X)ᵐᵒᵖ :=
-  { starRingEquiv with commutes' := fun r => by simp [star_algebra_map] }
+/-- `star` as an `AlgEquiv` -/
+noncomputable def starHom : FreeAlgebra R X ≃ₐ[R] (FreeAlgebra R X)ᵐᵒᵖ :=
+  { starRingEquiv with commutes' := fun r => by {
+    rename_i inst src
+    simp_all only [Equiv.toFun_as_coe_apply, MulOpposite.algebraMap_apply]
+    apply Eq.refl } }
 #align free_algebra.star_hom FreeAlgebra.starHom
```

Comment:
I think we should try to work out what has gone wrong here, `simp` should be making much more progress than it does.

Reply:
That's not true. Something was wrong here, it's #3008.

If `simp` fails in mathlib4 but succeeds in mathlib3, that's a warning that someone dropped a simp lemma and we need to add it back. Try not to rewrite a proof until you're sure what's causing the old one to fail.

Reply:
@eric-wieser Even with #3008 merged in `fun r => by simp [star_algebraMap]` still does not solve the goal. So indeed nothing was wrong here.

Reply:
What goal is left? That might mean there are more simp lemmas that are missing.

The `rfl` proof is good, but if `simp` is failing here it might also fail in another file where finding a replacement proof is much harder.

What's the remaining goal with #3008 merged and that line?